astropy/config/tests/test_configs.py

```diff
@@ -38,18 +38,125 @@ def test_paths():
 
 
 @pytest.mark.parametrize(
-    "environment_variable,func",
+    "environment_variable, cls, func",
     [
         # Regression test for #17514 - XDG_CACHE_HOME had no effect
-        pytest.param("XDG_CACHE_HOME", paths.get_cache_dir_path, id="cache"),
-        pytest.param("XDG_CONFIG_HOME", paths.get_config_dir_path, id="config"),
+        pytest.param(
+            "XDG_CACHE_HOME",
+            paths.set_temp_cache,
+            paths.get_cache_dir_path,
+            id="xdg-cache",
+        ),
```

Comment:
Would it be better to parametrize twice: once over `"ASTROPY"` and `"XDG"` and once over `"cache"` and `"config"`?

Reply:
Not sure. I just pushed a second commit to that effect. Let me know if you like it better.